app/Models/SizeVariant.php

```diff
@@ -183,39 +181,68 @@ public function sym_links(): HasManyBidirectionally
 	 */
 	public function getUrlAttribute(): string
 	{
-		$imageDisk = AbstractSizeVariantNamingStrategy::getImageDisk();
+		$imageDisk = Storage::disk($this->storage_disk->value);
 
 		if (
-			(Auth::user()?->may_administrate === true && !Configs::getValueAsBool('SL_for_admin')) ||
-			!Configs::getValueAsBool('SL_enable')
+			!Configs::getValueAsBool('SL_enable') ||
+			(!Configs::getValueAsBool('SL_for_admin') && Auth::user()?->may_administrate === true)
 		) {
 			return $imageDisk->url($this->short_path);
 		}
 
+		$storageAdapter = $imageDisk->getAdapter();
+		if ($storageAdapter instanceof AwsS3V3Adapter) {
+			return $this->getAwsUrl();
+		}
+
+		if ($storageAdapter instanceof LocalFilesystemAdapter) {
+			return $this->getSymLinkUrl();
+		}
+
+		throw new ConfigurationException('the chosen storage adapter "' . get_class($storageAdapter) . '" does not support the symbolic linking feature');
+	}
+
+	/**
+	 * Retrieve the tempary url from AWS if possible.
+	 *
+	 * @return string
+	 */
+	private function getAwsUrl(): string
+	{
 		// In order to allow a grace period, we create a new symbolic link,
-		// if the most recent existing link has reached 2/3 of its lifetime
 		$maxLifetime = Configs::getValueAsInt('SL_life_time_days') * 24 * 60 * 60;
-		$gracePeriod = $maxLifetime / 3;
+		$imageDisk = Storage::disk($this->storage_disk->value);
 
-		$storageAdapter = $imageDisk->getAdapter();
+		// Return the public URL in case the S3 bucket is set to public, otherwise generate a temporary URL
+		$visibility = config('filesystems.disks.s3.visibility', 'private');
+		// $visibility = $imageDisk->getConfig()['visibility'] ?? 'private';
+		if ($visibility === 'public') {
+			return $imageDisk->url($this->short_path);
```

Comment:
I guess at some point we'll need to document that the bucket has to be private to used the symlink options.

Reply:
Indeed.